### PR TITLE
Add org.gnome.Evolution

### PR DESCRIPTION
**Pull request by Sesivany**
On the behalf of my team which is working on the upstream Evolution I'd like to submit Evolution to Flathub.

### CODE REVIEW

File: org.gnome.Evolution.json [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+				{
+					"type": "archive",
+					"url": "http://www.openldap.org/software/download/OpenLDAP/openldap-release/openldap-2.4.45.tgz",
+					"sha256": "cdd6cffdebcd95161a73305ec13fc7a78e9707b46ca9f84fb897cd5626df3824"
+				}
+			]
+		},
+
+		{
+			"name": "dconf",
```
Comment (bilelmoussaoui) [inline org.gnome.Evolution.json]: Dconf is not needed.

File: org.gnome.Evolution.json [hunk 2]
```diff
@@ -0,0 +1,381 @@
+{
+	"app-id": "org.gnome.Evolution",
+	"runtime": "org.gnome.Platform",
+	"runtime-version": "3.28",
+	"sdk": "org.gnome.Sdk",
+	"command": "evolution",
+	"rename-icon": "evolution",
+	"copy-icon": true,
+	"branch": "gnome-3-30",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: branch has to be "stable" for flathub, so just remove this.

File: org.gnome.Evolution.json [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+		"*.la",
+		"*.a"
+	],
+	"finish-args": [
+		"--share=ipc",
+		"--share=network",
+		"--socket=x11",
+		"--socket=wayland",
+		"--socket=pulseaudio",
+                "--talk-name=org.gnome.OnlineAccounts",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: indenting

File: org.gnome.Evolution.json [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+					"type": "archive",
+					"url": "https://downloads.sourceforge.net/project/gtkspell/3.0.9/gtkspell3-3.0.9.tar.xz",
+					"sha256": "a4f4a4a2789f7499563e26d96b22d8085222ebe278da47d026b2de782b8b4d26"
+				}
+			]
+		},
+
+		{
+			"name": "evolution",
+			"buildsystem": "cmake",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: does this work with cmake-ninja? it's a *lot* faster to build.

**Comment by nedrichards** [comment]
bot, build org.gnome.Evolution

**Comment by nedrichards** [comment]
@Sesivany great to see! can you give us the other people's github usernames for when this is accepted so we can add the whole Evo team?

File: org.gnome.Evolution.json [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+				{
+					"type": "archive",
+					"url": "https://download.gnome.org/sources/libgweather/3.24/libgweather-3.24.1.tar.xz",
+					"sha256": "1227316037b63dac4153031a22abccb3d2ac679a7409e29cd303eec499eb653c"
+				}
+			]
+		},
+
+		{
+			"name": "openldap",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: this doesn't compile properly on aarch64. it needs to be reconfigured in a similar manner to this: https://github.com/flathub/shared-modules/blob/08e9b9eead7a95139be5d1fbc24bea730fb78f7a/SDL/SDL_net-1.2.8.json

File: org.gnome.Evolution.json [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+					"url": "https://github.com/libical/libical/releases/download/v3.0.3/libical-3.0.3.tar.gz",
+					"sha256": "5b91eb8ad2d2dcada39d2f81d5e3ac15895823611dc7df91df39a35586f39241"
+				}
+			]
+		},
+
+		{
+			"name": "liboauth",
+			"config-opts": [ "--enable-nss" ],
+			"cleanup": [ "/share/man" ],
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Looks like something that should be in global cleanup array.

File: org.gnome.Evolution.json [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+		"--socket=wayland",
+		"--socket=pulseaudio",
+                "--talk-name=org.gnome.OnlineAccounts",
+                "--talk-name=org.freedesktop.Notifications",
+                "--talk-name=org.freedesktop.portal.Desktop"
+	],
+	"modules": [
+		{
+			"name": "libical",
+			"cleanup": [ "/lib/cmake" ],
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Looks like something that should be in global cleanup array.

File: org.gnome.Evolution.json [hunk 8]
```diff
@@ -0,0 +1,381 @@
...
+		},
+
+		{
+			"name": "geocode-glib",
+			"config-opts": [
+				"--disable-introspection",
+				"--disable-gtk-doc"
+			],
+			"cleanup": [
+				"/share/gtk-doc"
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Looks like something that should be in global cleanup array.

File: org.gnome.Evolution.json [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+				{
+					"type": "archive",
+					"url": "https://www.gnupg.org/ftp/gcrypt/libgpg-error/libgpg-error-1.27.tar.bz2",
+					"sha256": "4f93aac6fecb7da2b92871bb9ee33032be6a87b174f54abf8ddf0911a22d29d2"
+				}
+			]
+		},
+
+		{
+			"name": "libgcrypt",
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Does it need newer libgcrypt than is already available in the runtime?

File: org.gnome.Evolution.json [hunk 10]
```diff
@@ -0,0 +1,381 @@
...
+		},
+
+		{
+			"name": "evolution-ews",
+			"buildsystem": "cmake",
+			"sources": [
+				{
+					"type": "git",
+					"url": "https://gitlab.gnome.org/GNOME/evolution-ews.git",
+					"branch": "gnome-3-30"
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Is this a tag or a branch? If the latter, specify also a commit hash for builds reproducibility.

File: org.gnome.Evolution.json [hunk 11]
```diff
@@ -0,0 +1,381 @@
...
+			],
+			"cleanup": [
+				"/etc/xdg",
+				"/share/GConf"
+			],
+			"sources": [
+				{
+					"type": "git",
+					"url": "https://gitlab.gnome.org/GNOME/evolution.git",
+					"branch": "gnome-3-30"
```
Comment (barthalion) [inline org.gnome.Evolution.json]: Is this a tag or a branch? If the latter, specify also a commit hash for builds reproducibility.

File: org.gnome.Evolution.json (hunk 4 above)
Comment (Sesivany) [reply org.gnome.Evolution.json]: I tried to build it with cmake-ninja and it seems to work, so I'll add it to the next version of the manifest.

**Comment by Sesivany** [comment]
I updated the manifest based on feedback accommodating all suggestions and problems except for the cleanup, we still prefer doing it for each module separately.
Adding commit hashes is not very practical because it breaks the build after every commit to that git branch, but that's what we have now, with the next upstream release in 2 weeks we will have everything in tarballs and will change the manifest accordingly.

**Comment by AsciiWolf** [comment]
bot, build org.gnome.Evolution

**Comment by nedrichards** [comment]
@Sesivany I mean, that's the point of the reproducibility - for now just point to the commit hash that you know works, which will be stable and not go away, then when the tarballs are out switch to that. If you need to update the build here in the interim then you'll need to change the commits.

File: org.gnome.Evolution.json [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+        ],
+        "finish-args": [
+            "--share=ipc",
+            "--share=network",
+            "--socket=x11",
+            "--socket=wayland",
+            "--socket=pulseaudio",
+            "--talk-name=org.gnome.OnlineAccounts",
+            "--talk-name=org.freedesktop.Notifications",
+            "--talk-name=org.freedesktop.portal.Desktop"
```
Comment (TingPing) [inline org.gnome.Evolution.json]: This is a default permission.

File: org.gnome.Evolution.json [hunk 13]
```diff
@@ -0,0 +1,374 @@
...
+        ],
+        "modules": [
+            {
+                "name": "libical",
+                "cleanup": [ "/lib/cmake" ],
+                "buildsystem": "cmake-ninja",
+                "config-opts": [
+                    "-DCMAKE_INSTALL_LIBDIR:PATH=/app/lib",
```
Comment (TingPing) [inline org.gnome.Evolution.json]: Shouldn't be needed.

File: org.gnome.Evolution.json [hunk 14]
```diff
@@ -0,0 +1,374 @@
...
+                        "type": "archive",
+                        "url": "https://downloads.sourceforge.net/project/liboauth/liboauth-1.0.3.tar.gz",
+                        "sha256": "0df60157b052f0e774ade8a8bac59d6e8d4b464058cc55f9208d72e41156811f"
+                    }
+                ]
+            },
+     
+            {
+                "name": "gnome-online-accounts",
+                "config-opts": [ "--disable-Werror", "--disable-telepathy", "--disable-documentation", "--disable-backend" ],
```
Comment (TingPing) [inline org.gnome.Evolution.json]: Wrap your options on new lines.

File: org.gnome.Evolution.json (hunk 13 above)
Comment (Sesivany) [reply org.gnome.Evolution.json]: I'm not able to build Evolution after removing this module.

File: org.gnome.Evolution.json [hunk 15]
```diff
@@ -0,0 +1,375 @@
...
+                    "--disable-introspection",
+                    "--disable-vala"
+                ],
+                "cleanup": [
+                    "/share/gtk-doc"
+                ],
+                "sources": [
+                    {
+                        "type": "archive",
+                        "url": "https://download.gnome.org/sources/libgweather/3.24/libgweather-3.24.1.tar.xz",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: Does evo work with a more recent gweather? such as 3.28? https://download.gnome.org/sources/libgweather/3.28/

File: org.gnome.Evolution.json [hunk 16]
```diff
@@ -0,0 +1,375 @@
...
+            "/share/aclocal",
+            "*.la",
+            "*.a"
+        ],
+        "finish-args": [
+            "--share=ipc",
+            "--share=network",
+            "--socket=x11",
+            "--socket=wayland",
+	    "--socket=pulseaudio",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: indentation

File: org.gnome.Evolution.json [hunk 17]
```diff
@@ -0,0 +1,375 @@
...
+                        "sha256": "0df60157b052f0e774ade8a8bac59d6e8d4b464058cc55f9208d72e41156811f"
+                    }
+                ]
+            },
+     
+            {
+                "name": "gnome-online-accounts",
+                "config-opts": 
+		[ 
+		"--disable-Werror",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: indentation

File: org.gnome.Evolution.json [hunk 18]
```diff
@@ -0,0 +1,375 @@
...
+                    "--disable-introspection",
+                    "--disable-gtk-doc"
+                ],
+                "cleanup": [
+                    "/share/gtk-doc"
+                ],
+                "sources": [
+                    {
+                        "type": "archive",
+                        "url": "https://download.gnome.org/sources/geocode-glib/3.24/geocode-glib-3.24.0.tar.xz",
```
Comment (nedrichards) [inline org.gnome.Evolution.json]: Does evo work with a more recent geocode-glib? https://download.gnome.org/sources/geocode-glib/3.26/

Comment (Sesivany) [reply org.gnome.Evolution.json]: It seems to work just fine with the newer version, I'm going to update the manifest.

**Comment by barthalion** [comment]
bot, build org.gnome.Evolution

**Comment by alexlarsson** [comment]
I took a look at this, and overall it looks ok, although some cleanups (/share/gtk-doc and /share/man) could be made global to avoid repeating. But one thing stands out. 

Why do you bundle dconf, and does dconf access actually work in evo? We already ship a copy of dconf in the runtime, and that one has been lightly modified to work better in a sandbox. But in addition to having the code there you need to set some permissions to be able to actually access dconf: http://docs.flatpak.org/en/latest/sandbox-permissions.html#dconf-access

**Comment by alexlarsson** [comment]
bot, build org.gnome.Evolution

**Comment by alexlarsson** [comment]
Merged, new repo here: https://github.com/flathub/org.gnome.Evolution

**Pull request closed without merging**